### PR TITLE
Improve unity catalog profiler speed part 2

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Running `ANALYZE TABLE` every time is too costly, add config flag to en/disable it

### 🤓 What?

- Creat a connection pool after getting all the table
- Don't run `ANALYZE` by default
- Only profile tables that the table type is in the white list

### 🧪 Tested?

Test against prod env

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/unity_catalog/profile/README.md [hunk 1]
```diff
@@ -40,6 +40,15 @@ The max number of concurrent queries to the databricks compute node can be confi
 max_concurrency: <max_number_of_queries> # Default to 10
 ```
 
+#### Analyze table
+
+To run `ANALYZE TABLE` query before the connector getting the statistics, or only for tables have no statistics.
+
+```yaml
+analyze: true # Default is false
```
Comment (mars-lan) [inline metaphor/unity_catalog/profile/README.md:48]: Not sure why you need 2 flags? I can only think of 3 situations:

1. Has pre-cached statistics available, return immediately (ignore config)
2. No pre-cached statistics available, run analyze (`analyze_if_no_statistics` = true)
3. No pre-cached statistics available, skip analyze (`analyze_if_no_statistics` = false)

Comment (elic-eon) [reply metaphor/unity_catalog/profile/README.md:48]: 1. We can still run analyze if we want to update the cached statistics
2. run analyze
3. skip analyze, getting null value

Comment (elic-eon) [reply metaphor/unity_catalog/profile/README.md:48]: Basically, with `analyze=true` will behave like previous implementation.

Comment (mars-lan) [reply metaphor/unity_catalog/profile/README.md:48]: Also update README?

File: metaphor/unity_catalog/profile/extractor.py [hunk 2]
```diff
@@ -130,9 +148,11 @@ def _get_tables(self) -> List[TableInfo]:
                 ]
                 for table_info in table_infos:
                     table_type = table_info.table_type
-                    if table_type is None or table_type in (
-                        TableType.VIEW,
-                        TableType.MATERIALIZED_VIEW,
+                    if table_type not in (
+                        TableType.EXTERNAL,
```
Comment (mars-lan) [inline metaphor/unity_catalog/profile/extractor.py:152]: Let's define this as a const at the top for easier modification in the future.

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM

**Review by mars-lan — APPROVED** [review]

**Pull request merged**